modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableIndexStoragesSupplier.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.table.distributed;
+
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Supplier table index storages.
+ */
+public interface TableIndexStoragesSupplier {
```

Comment:
What's the reason of calling it a supplier? Design of this interface look peculiar, I guess it's temporary

Reply:
I could not find the correct name, I assume that it is temporary, when catalogs appear, we will get rid of this code.